networkit/cpp/graph/Graph.cpp

```diff
@@ -1041,21 +1040,43 @@ std::vector<node> Graph::neighbors(node u) const {
 Graph Graph::transpose() const {
 	if (directed == false) {
 		throw std::runtime_error("The transpose of an undirected graph is "
-		                         "identical to the original graph.");
+								 "identical to the original graph.");
 	}
 
-	GraphBuilder gB(z, weighted, true);
+	Graph GTranspose(z, weighted, true);
 
-	this->forEdges([&](node u, node v) { gB.addHalfEdge(v, u, weight(u, v)); });
+	// prepare edge id storage if input has indexed edges
+	GTranspose.edgesIndexed = edgesIndexed;
+	GTranspose.omega = omega;
+	GTranspose.inEdgeIds.resize(z);
+	GTranspose.outEdgeIds.resize(z);
```

Comment:
Shouldn't the resize only happen in the case that edge IDs exist?

Reply:
Yes, you're right!